stubs/docutils/docutils/parsers/__init__.pyi

```diff
@@ -1,14 +1,15 @@
 from typing import Any, ClassVar
 
 from docutils import Component
+from docutils.nodes import document
 
 class Parser(Component):
     component_type: ClassVar[str]
     config_section: ClassVar[str]
     inputstring: Any  # defined after call to setup_parse()
     document: Any  # defined after call to setup_parse()
-    def parse(self, inputstring: str, document) -> None: ...
-    def setup_parse(self, inputstring: str, document) -> None: ...
+    def parse(self, inputstring: str, document: document) -> None: ...
```

Comment:
Can you rename the type to `_document`? Under normal scoping rules `document` in the annotation would refer to the class attribute defined above. Not sure why CI doesn't currently complain.